src/mower_logic/src/monitoring/monitoring.cpp

```diff
@@ -125,114 +113,119 @@ void pose_received(const xbot_msgs::AbsolutePose::ConstPtr &msg) {
-  mow_motor_temp_data_pub =
-      n->advertise<xbot_msgs::SensorDataDouble>("xbot_monitoring/sensors/" + si_mow_motor_temp.sensor_id + "/data", 10);
-  si_mow_motor_temp_pub.publish(si_mow_motor_temp);
-
-  si_mow_motor_current.sensor_id = "om_mow_motor_current";
-  si_mow_motor_current.sensor_name = "Mow Motor Current";
-  si_mow_motor_current.value_type = xbot_msgs::SensorInfo::TYPE_DOUBLE;
-  si_mow_motor_current.value_description = xbot_msgs::SensorInfo::VALUE_DESCRIPTION_CURRENT;
-  si_mow_motor_current.unit = "A";
-  si_mow_motor_current_pub = n->advertise<xbot_msgs::SensorInfo>(
-      "xbot_monitoring/sensors/" + si_mow_motor_current.sensor_id + "/info", 1, true);
-  mow_motor_current_data_pub = n->advertise<xbot_msgs::SensorDataDouble>(
-      "xbot_monitoring/sensors/" + si_mow_motor_current.sensor_id + "/data", 10);
-  si_mow_motor_current_pub.publish(si_mow_motor_current);
-
-  si_gps_accuracy.sensor_id = "om_gps_accuracy";
-  si_gps_accuracy.sensor_name = "GPS Accuracy";
-  si_gps_accuracy.value_type = xbot_msgs::SensorInfo::TYPE_DOUBLE;
-  si_gps_accuracy.value_description = xbot_msgs::SensorInfo::VALUE_DESCRIPTION_DISTANCE;
-  si_gps_accuracy.unit = "m";
-  si_gps_accuracy_pub =
-      n->advertise<xbot_msgs::SensorInfo>("xbot_monitoring/sensors/" + si_gps_accuracy.sensor_id + "/info", 1, true);
-  gps_accuracy_data_pub =
-      n->advertise<xbot_msgs::SensorDataDouble>("xbot_monitoring/sensors/" + si_gps_accuracy.sensor_id + "/data", 10);
-  si_gps_accuracy_pub.publish(si_gps_accuracy);
+  for (auto &sc_pair : sensor_configs) {
+    if (sc_pair.second.existCB != nullptr && !sc_pair.second.existCB()) {
+      ROS_INFO_STREAM("Skipped monitoring of sensor " << sc_pair.first);
+      continue;
+    }
+
+    sc_pair.second.si.sensor_id = sc_pair.first;
+    sc_pair.second.si.sensor_name = sc_pair.second.name;
+
+    sc_pair.second.si.unit = sc_pair.second.unit;
+    sc_pair.second.si.value_type = xbot_msgs::SensorInfo::TYPE_DOUBLE;
+    sc_pair.second.si.value_description = sc_pair.second.value_desc;
+
+    // Set sensor threshold values
+    set_sensor_limits(sc_pair.second);
+
+    sc_pair.second.si_pub =
+        n->advertise<xbot_msgs::SensorInfo>("xbot_monitoring/sensors/" + sc_pair.first + "/info", 1, true);
+    sc_pair.second.data_pub =
+        n->advertise<xbot_msgs::SensorDataDouble>("xbot_monitoring/sensors/" + sc_pair.first + "/data", 10);
+    sc_pair.second.si_pub.publish(sc_pair.second.si);
+  }
+}
+
+void reconfigCB(const mower_logic::MowerLogicConfig &config) {
+  ROS_INFO_STREAM("Monitoring received new mower_logic config");
+  mower_logic_config = config;
+
+  // Set and publish new sensor limits
+  for (auto &sc_pair : sensor_configs) {
+    // Skip if sensor doesn't exists
+    if (sc_pair.second.existCB != nullptr && !sc_pair.second.existCB()) continue;
+
+    set_sensor_limits(sc_pair.second);
+    // FIXME: xbot_monitoring need to be adapted to re-read sensors/*/info
```

Comment:
I think that is done?

Reply:
You mean the FIXME comment? Yes, that's done. Will remove...